draft-ietf-quic-transport.md

```diff
@@ -542,68 +545,138 @@ the server supports, as shown below.
 
-* Octet 0: 0x83
-* Octets 1-8: Connection ID (echoed)
-* Octets 9-12: Packet Number
-* Octets 13-16: Version
-* Octets 17+: Payload
+### Client Initial Packet {#packet-client-initial}
 
-Final Server Cleartext packets contains:
+The Client Initial packet has a type octet of 0x82.  It carries the first
+cryptographic handshake message sent by the client.
 
-* Octet 0: 0x84
-* Octets 1-8: Connection ID (final)
-* Octets 9-12: Packet Number
-* Octets 13-16: Version
-* Octets 17+: Payload
+The client populates the connection ID field with randomly selected values,
+unless it has received a packet from the server.  If the client has received a
+packet from the server, the connection ID field uses the value provided by the
+server.
 
-The client MUST choose a random 64-bit value and use it as the initial
-Connection ID in all packets until the server replies with the final Connection
-ID. The server echoes the client's Connection ID in Non-Final Server Cleartext
-packets.  The first Final Server Cleartext and all subsequent packets MUST use
-the final Connection ID, as described in {{connection-id}}.
+The packet number used for Client Initial packets is initialized with a random
+value each time the new contents are created for the packet.  Retransmissions of
+the packet contents increment the packet number by one, see
+({{packet-numbers}}).
 
-The payload of a Cleartext packet consists of a sequence of frames, as described
-in {{frames}}.  Cleartext packets also include an integrity check, described in
-{{QUIC-TLS}}.
+The payload of a Client Initial packet consists of a STREAM frame (or frames)
+for stream 0 containing a cryptographic handshake message, plus any PADDING
+frames necessary to ensure that the packet is at least the minimum size (see
+{{packetization}}).  This stream frame always starts at an offset of 0 (see
+{{stateless-retry}}).
+
+The client uses the Client Initial Packet type for any packet that contains an
+initial cryptographic handshake message.  This includes all cases where a new
+packet containing the initial cryptographic message needs to be created, this
+includes the packets sent after receiving a Version Negotiation
+({{packet-version}}) or Server Stateless Retry packet
+({{packet-server-stateless}}).
+
+
+### Server Stateless Retry Packet {#packet-server-stateless}
+
+A Server Stateless Retry packet has a type octet of 0x83.  It carries
+cryptographic handshake messages and acknowledgments.  It is used by a server
+that wishes to perform a stateless retry (see {{stateless-retry}}).
+
+The connection ID field in a Server Stateless Retry packet contains a server
+selected value, see {{connection-id}}.
+
+The packet number field echoes the packet number of the triggering client
+packet.  This allows a client to verify that the server received its packet.
+
+After receiving a Server Stateless Retry packet, the client MUST generate a new
+Client Initial packet containing the next cryptographic handshake message.  The
```

Comment:
I think we should state that the client MUST reset all connection state at this point, simplifying this description. ... and you do say it below, but I'd just say that and drop the rest of this text that describes what is reset. I would suggest adding that if a client has received an ACK frame thus far from the server, it MAY use the RTT sample to seed the new connection's RTT estimator.